queries/d/highlights.scm

```diff
@@ -35,254 +86,194 @@
   "$"
 ] @punctuation.special
 
-;; Constants
+;; Keywords
 
-[
-  "__FILE_FULL_PATH__"
-  "__FILE__"
-  "__FUNCTION__"
-  "__LINE__"
-  "__MODULE__"
-  "__PRETTY_FUNCTION__"
-] @constant.macro
+; these are listed first, because they override keyword queries
+(identity_expression (in) @keyword.operator)
+(identity_expression (is) @keyword.operator)
 
 [
-  (wysiwyg_string)
-  (alternate_wysiwyg_string)
-  (double_quoted_string)
-  (hex_string)
-  (delimited_string)
-  (token_string)
-] @string
-
-(character_literal) @character
-
-(integer_literal) @number
-
-(float_literal) @float
+  (not_in)
+  (not_is)
+] @keyword.operator
 
 [
-  "true"
-  "false"
-] @boolean
-
-;; Functions
-
-(func_declarator
-  (identifier) @function
-)
+  (const)
+  (immutable)
+  (inout)
+  (shared)
+] @type.qualifier
 
 [
-  "__traits"
-  "__vector"
-  "assert"
-  "is"
-  "mixin"
-  "pragma"
-  "typeid"
-] @function.builtin
-
-(import_expression
-  "import" @function.builtin
-)
-
-(parameter
-  (var_declarator
-    (identifier) @parameter
-  )
-)
-
-(function_literal
-  (identifier) @parameter
-)
-
-(constructor
-  "this" @constructor
-)
-
-(destructor
-  "this" @constructor
-)
+  (gshared)
+  (scope)
+  (abstract)
+  (final)
+  (override)
+  (lazy)
+  (align)
+  (extern)
+  (static)
+  (synchronized)
+  (auto)
+  (ref)
+  (nothrow)
+  (pure)
+  (deprecated)
+] @storageclass
 
-;; Keywords
+(parameter_attribute (return) @storageclass)
+(parameter_attribute (in) @storageclass)
+(parameter_attribute (out) @storageclass)
 
 [
-  "case"
-  "default"
-  "else"
-  "if"
-  "switch"
+  (case)
+  (default)
+  (if)
+  (else)
+  (switch)
 ] @conditional
 
 [
-  "break"
-  "continue"
-  "do"
-  "for"
-  "foreach"
-  "foreach_reverse"
-  "while"
+  (break)
+  (continue)
+  (do)
+  (for)
+  (while)
+  (foreach)
+  (foreach_reverse)
 ] @repeat
 
 [
-  "__parameters"
-  "alias"
-  "align"
-  "asm"
-  "auto"
-  "body"
-  "class"
-  "debug"
-  "enum"
-  "export"
-  "goto"
-  "interface"
-  "invariant"
-  "macro"
-  "out"
-  "override"
-  "package"
-  "static"
-  "struct"
-  "template"
-  "union"
-  "unittest"
-  "version"
-  "with"
-] @keyword
-
-[
-  "delegate"
-  "function"
+  (delegate)
+  (function)
 ] @keyword.function
 
-"return" @keyword.return
+(return) @keyword.return
 
-[
-  "cast"
-  "new"
-] @keyword.operator
+(debug) @debug
+(import) @include
 
 [
-  "+"
-  "++"
-  "+="
-  "-"
-  "--"
-  "-="
-  "*"
-  "*="
-  "%"
-  "%="
-  "^"
-  "^="
-  "^^"
-  "^^="
-  "/"
-  "/="
-  "|"
-  "|="
-  "||"
-  "~"
-  "~="
-  "="
-  "=="
-  "=>"
-  "<"
-  "<="
-  "<<"
-  "<<="
-  ">"
-  ">="
-  ">>"
-  ">>="
-  ">>>"
-  ">>>="
-  "!"
-  "!="
-  "&"
-  "&&"
-] @operator
-
-[
-  "catch"
-  "finally"
-  "throw"
-  "try"
-] @exception
```

Comment:
The @exception selector is missing in the other editor plugins I've done -- but it would be easy to fix that here by moving those words into a block from where they are located.

Reply:
Oooh, this looks like an error of omission.

Reply:
Added. Thanks for noticing!